src/app/pdok/lookup-object/lookup-object.component.ts

```diff
@@ -29,37 +32,66 @@ export class LookupObjectComponent implements OnInit {
+      this.lookupObjectForm.addControl('postcode', new FormControl(this.lookupObject.postcode, [Validators.required]));
+      this.lookupObjectForm.addControl('woonplaatsnaam', new FormControl(this.lookupObject.woonplaatsnaam, [Validators.required]));
+    }
+
+    if (this.lookupObject.type == 'weg') {
+      this.lookupObjectForm.addControl('gemeentenaam', new FormControl(this.lookupObject.gemeentenaam, [Validators.required]));      
+      this.lookupObjectForm.addControl('straatnaam', new FormControl(this.lookupObject.straatnaam, [Validators.required]));
+      this.lookupObjectForm.addControl('woonplaatsnaam', new FormControl(this.lookupObject.woonplaatsnaam, [Validators.required]));
+    }
+
+    if (this.lookupObject.type === 'gemeente') {
+      this.lookupObjectForm.addControl('gemeentenaam', new FormControl(this.lookupObject.gemeentenaam, [Validators.required]));
+    }
+
+    if (this.lookupObject.type === 'woonplaats') {
+      this.lookupObjectForm.addControl('gemeentenaam', new FormControl(this.lookupObject.gemeentenaam, [Validators.required]));      
+      this.lookupObjectForm.addControl('woonplaatsnaam', new FormControl(this.lookupObject.woonplaatsnaam, [Validators.required]));
+    }
+
+    if (this.lookupObject.type === 'postcode') {
+      this.lookupObjectForm.addControl('straatnaam', new FormControl(this.lookupObject.straatnaam, [Validators.required]));
+      this.lookupObjectForm.addControl('postcode', new FormControl(this.lookupObject.postcode, [Validators.required]));
+      this.lookupObjectForm.addControl('woonplaatsnaam', new FormControl(this.lookupObject.woonplaatsnaam, [Validators.required]));
+    }
+
+    this.lookupObjectForm.addControl('aliassen', new FormControl(this.lookupObject.aliassen));
   }
 
   onSaveClicked() {
+    let newLookUpObject: LookupObject;
     console.log('save');
     if (this.lookupObjectForm.valid) {
-      //debugger;
       this.saving = true;
-      this.lookupObject = Object.assign(this.lookupObject, this.lookupObjectForm.value);
-      this.lookupService.updateFakeLookup(this.lookupObject).then((result) => {
-        console.log('lookupSaved', result);
-        this.lookupObjectForm.patchValue(result); //needed for weergavenaam
-        /*
-        this.lookupObjectForm.patchValue({
-          weergavenaam: result.weergavenaam
+      newLookUpObject = Object.assign({}, this.lookupObject, this.lookupObjectForm.value);
+      this.lookupService.updateFakeLookup(newLookUpObject)
+        .then((result) => {
+          console.log('lookupSaved', result);
+          this.lookupObjectForm.patchValue(result); //needed for weergavenaam
+          this.openSnackBar('Het object is succesvol bijgewerkt', 'ok');
+          this.saving = false;
+          this.saved.emit(result);
+        })
+        .catch((error) => {
+          this.openSnackBar('Er is iets misgegaan, probeer het later nog eens', 'ok', 5000);
```

Comment:
The error messages should come from the backend/service. They shouldn't be defined in the component